netket/stats/mc_stats.py

```diff
@@ -98,42 +101,87 @@ def __getattr__(self, name):
         elif name in ("variance", "Variance"):
             return self.variance
         elif name in ("error_of_mean", "Sigma"):
-            return self.error_of_mean
+            return self.error
         elif name in ("R_hat", "R"):
-            return self.R_hat
+            return self.r_hat
         elif name in ("tau_corr", "TauCorr"):
             return self.tau_corr
         else:
             raise AttributeError(
                 "'Stats' object object has no attribute '{}'".format(name)
             )
 
+    @property
+    def n_chains(self):
+        return self._μ.size * mpi.n_nodes
 
-def _get_blocks(data, block_size):
-    chain_length = data.shape[1]
+    @struct.property_cached(pytree_node=True)
+    def mean(self) -> float:
+        return _mean(self._μ)
 
-    n_blocks = int(np.floor(chain_length / float(block_size)))
+    @struct.property_cached(pytree_node=True)
+    def variance(self) -> float:
+        return _mean(self._μ2) - self.mean ** 2
 
-    return data[:, 0 : n_blocks * block_size].reshape((-1, block_size)).mean(axis=1)
+    @struct.property_cached(pytree_node=True)
+    def error(self) -> float:
+        return jnp.sqrt(self.batch_variance / self.n_chains)
 
+    @struct.property_cached(pytree_node=True)
+    def r_hat(self) -> float:
```

Comment:
If you want to rename this , may I suggest `rhat`? The underscore version is less convenient to type.

Reply:
I'd like to keep this PR only about the method itself, let's change the syntax in another PR so we can discuss that more focusedly, because it might be fairly breaking... ok?